internal/controller/etcdcluster_controller.go

```diff
@@ -67,22 +70,64 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		// Error retrieving object, requeue
 		return reconcile.Result{}, err
 	}
+
 	// If object is being deleted, skipping reconciliation
 	if !instance.DeletionTimestamp.IsZero() {
 		return reconcile.Result{}, nil
 	}
 
-	// fill conditions
-	if len(instance.Status.Conditions) == 0 {
-		factory.FillConditions(instance)
-	}
-
-	// ensure managed resources
+	// Ensure managed resources
 	if err := r.ensureClusterObjects(ctx, instance); err != nil {
 		logger.Error(err, "cannot create Cluster auxiliary objects")
 		return r.updateStatusOnErr(ctx, instance, fmt.Errorf("cannot create Cluster auxiliary objects: %w", err))
 	}
 
+	// Check if StatefulSet exists
+	statefulSet := &appsv1.StatefulSet{}
+	err = r.Get(ctx, client.ObjectKey{Name: instance.Name, Namespace: instance.Namespace}, statefulSet)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.V(2).Info("StatefulSet not found, skipping storage check", "namespaced_name", req.NamespacedName)
+		} else {
+			return reconcile.Result{}, err
+		}
+	} else {
+		// Check if storage size has changed
+		if len(statefulSet.Spec.VolumeClaimTemplates) > 0 {
+			pvcName := statefulSet.Spec.VolumeClaimTemplates[0].Name
+			pvc := &corev1.PersistentVolumeClaim{}
+			err = r.Get(ctx, client.ObjectKey{Name: pvcName, Namespace: instance.Namespace}, pvc)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					logger.V(2).Info("PVC not found, skipping storage check", "pvc_name", pvcName)
+				} else {
+					return reconcile.Result{}, err
+				}
+			} else {
+				desiredStorage := instance.Spec.Storage.VolumeClaimTemplate.Spec.Resources.Requests[corev1.ResourceStorage]
+				currentStorage := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
you have to compare storage from `statefulSet.Spec.VolumeClaimTemplate`, not pvc